### PR TITLE
simplified batch softmax

### PR DESCRIPTION
**Pull request by taras-sereda**

### CODE REVIEW

**Comment by taras-sereda** [comment]
Hi @SeanNaren! pytorch allows now to apply softmax across the batch by specifying necessary dim.  take a look. now the code is more concise.

**Comment by SeanNaren** [comment]
LGTM thanks!
